### ISSUE TITLE
[BUG]:  Vampire Night / Ninja Assault : Guncon2 auto reload not expected

### ISSUE THREAD

**Issue by ginzu2077** [PCSX2/pcsx2#8818]
### Describe the Bug

the lightgun automatically reloads after a few shots without any user action and disrupts gameplay or the shooting action sometimes launches a reload more accurately. 

Impacted games : Vampire night and Ninja Assault. (all versions). US/JPN/EU

### Reproduction Steps

Launch one of the following game
as soon as the game arcade mode or Arcade story start , just shot as much as you can.

[Ninja Assault]
SLPS-20218 JPN  (failed to pass the calibration screen)
SLUS-20492 NTSC-US
SCES-50889 PAL-M Europe-Australia

[Vampire Night]
SCES-50411 PAL-M Europe-Australia
SLUS-20221 NTSC-US

### Expected Behavior
(select guncon2 on usb port 1 under controllers) tested with a gun4ir
when you think you are going to fire , a reload is launched at an unexpected moment. (happens often)

### PCSX2 Revision

1.7.4482

### Operating System

Windows 10 (64bit)

### CPU

i7-9700K

### GPU

GTX 970

### GS Settings

n/a  i use default vulkan renderer , selected my adapter : nvidia 970 gtx

**Comment by ginzu2077** [comment]
As a side note, i can now tell there is also an input lag with the game vampire night. 
Maybe there is a little input lag with Ninja Assault too but it is less obvious than in Vampire night.

**Comment by ginzu2077** [comment]
> Try the given workaround in [#7803 (comment)](https://github.com/PCSX2/pcsx2/issues/7803#issuecomment-1507362961) it works really well across the gun games.

Thank you, but can you explain how to access this file ? i can't find it.

**Comment by Tovarichtch** [comment]
> > Try the given workaround in [#7803 (comment)](https://github.com/PCSX2/pcsx2/issues/7803#issuecomment-1507362961) it works really well across the gun games.
> 
> Thank you, but can you explain how to access this file ? i can't find it.

You need to modify the file and compile it afterwards. But it just got PR : https://github.com/PCSX2/pcsx2/pull/8828

Give it a try in the next nightly build!

**Comment by ginzu2077** [comment]
Ok, thank you.  i updated to the last version...  
But now,  i can't fire any shot in Vampire Night....  
I tried TC2 and  i can fire shots without problems.

**Comment by Mrlinkwii** [comment]
@ginzu2077 so the issue is solved on latest nightly

**Comment by ginzu2077** [comment]
I use the updater... do you think i have to download the latest one from scratch ?

**Comment by ginzu2077** [comment]
Hello ,

I tested the  workaround you mentionned above : https://github.com/PCSX2/pcsx2/issues/7803
                                                                                 https://github.com/PCSX2/pcsx2/pull/8828

I tested successfully all my games! NO more input lag at all! 👍
BUT :  it's mandatory to adjust each profile for each games and we have 2 stuffs to do to make it to work properly  :
1) Do not set any button on the Offscreen reload!  -->  use button C or button A instead!
2) Calibration button should no longer be the same as the trigger button! i use middle button.
The only problem for people who have a gun with only 1 or 2 buttons is they need to use a third button when they do the calibration. so, the perfect gun should have at least 3 buttons. (1: trigger,2:reload,3 calibration)

I no longer have reload problems like before since this patch! So, i can tell the issue is fixed!

great job guys! i just hope it will not break in a further update of the Nightly build.
